src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,211 @@
+        t = shiftl(st(2), 17)
+        st(3) = ieor(st(3), st(1))
+        st(4) = ieor(st(4), st(2))
+        st(2) = ieor(st(2), st(3))
+        st(1) = ieor(st(1), st(4))
+        st(3) = ieor(st(3), t)
+        st(4) = rol64(st(4), 45)
+        return
+    end function xoshiro256ss
+
+    function rol64(x, k) result(res)
+        integer(int64), intent(in) :: x
+        integer, intent(in) :: k
+        integer(int64) :: t1, t2, res
+
+        t1 = shiftr(x, (64 - k))
+        t2 = shiftl(x, k)
+        res = ior(t1, t2)
+        return
+    end function rol64
+
+
+    subroutine jump
+    ! This is the jump function for the xoshiro256ss generator. It is equivalent
+    ! to 2^128 calls to xoshiro256ss(); it can be used to generate 2^128
+    ! non-overlapping subsequences for parallel computations.
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    integer(int64) :: jp(4) = [1733541517147835066_int64,                       &
+                               -3051731464161248980_int64,                      &
+                               -6244198995065845334_int64,                      &
+                               4155657270789760540_int64]
+    integer(int64) :: s1 = 0, s2 = 0, s3 = 0, s4 = 0, c = 1_int64
+    integer :: i, j, k
+
+    do i = 1, 4
+        do j = 1, 64
+            if(iand(jp(i), shiftl(c, j - 1)) /= 0) then
+                s1 = ieor(s1, st(1))
+                s2 = ieor(s2, st(2))
+                s3 = ieor(s3, st(3))
+                s4 = ieor(s4, st(4))
+            end if
+            k = xoshiro256ss( )
+        end do
+    end do
+    st(1) = s1
+    st(2) = s2
+    st(3) = s3
+    st(4) = s4
+    end subroutine jump
+
+    subroutine long_jump
+    ! This is the long-jump function for the xoshiro256ss generator. It is
+    ! equivalent to 2^192 calls to xoshiro256ss(); it can be used to generate
+    ! 2^64 starting points, from each of which jump() will generate 2^64
+    ! non-overlapping subsequences for parallel distributed computations
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    integer(int64) :: jp(4) = [8566230491382795199_int64,                       &
+                               -4251311993797857357_int64,                      &
+                               8606660816089834049_int64,                       &
+                               4111957640723818037_int64]
+    integer(int64) :: s1 = 0, s2 = 0, s3 = 0, s4 = 0, c = 1_int64
+    integer(int32) :: i, j, k
+
+    do i = 1, 4
+        do j = 1, 64
+            if(iand(jp(i), shiftl(c, j - 1)) /= 0) then
+                s1 = ieor(s1, st(1))
+                s2 = ieor(s2, st(2))
+                s3 = ieor(s3, st(3))
+                s4 = ieor(s4, st(4))
+            end if
+            k = xoshiro256ss()
```

Comment:
Where is `k` used in this subroutine?

Reply:
k is never used. But calling xoshiro256ss function will change the saved state randomly.

Reply:
I suggest to add a comment to explain the purpose of this call, e.g.,
```fortran
!xoshiro256ss must be called to change the saved state randomly
```

Reply:
I am not sure to add the comment. It is my guess that function call change the state randomly. The original authors had never mentioned.